net/dns/resolver/tsdns.go

```diff
@@ -596,10 +613,29 @@ func (r *Resolver) resolveLocal(domain dnsname.FQDN, typ dns.Type) (netip.Addr,
 
 	r.mu.Lock()
 	hosts := r.hostToIP
+	wildcards := r.wildcards
 	localDomains := r.localDomains
 	r.mu.Unlock()
 
-	addrs, found := hosts[domain]
+	var found = false
+	var addrs []netip.Addr
+
+	addrs, found = hosts[domain]
+	if !found {
+		// Find matching subdomain entries
+		for _, wildcard := range wildcards {
+			// Entries in wcHosts already have a trailing '.', and Contains adds another
+			fqdn, err := dnsname.ToFQDN(string(wildcard))
```

Comment:
we shouldn't need to be doing any allocation at serving time here.

let's get it into a form in memory that's efficient to serve from. The O(n) loop over wildcards is already slightly shady (because I don't know how many wildcards users will end up having... 0-few usually, but what's the upper bound? hundreds? thousand or more?)